### PR TITLE
Add amount input component to fix input bugs

### LINKED ISSUE #536: Faucet: Amount input bug + general amount input component
There are some bugs in the faucet amount input:

- Once a number has been entered, the first digit of the input cannot be removed
- The step parameter of the input allows for three decimal places using the increment button, but only integers can be typed in directly

Question: should the faucet allow only integer amounts, or should it allow to the max decimal places allowed for the token (6 for NAM)? I would guess for simplicity that integer amounts is fine.

More generally, it would be good to have an amount input component that we can reuse everywhere we need accept user input for amounts.

EDIT: Looks like the interface send token form also has a buggy amount input.

### PR DESCRIPTION
**Pull request by emccorson**
Fixes input amount bugs in faucet and interface send token form.

Fixes #536.

### CODE REVIEW

**Review by pedrorezende — APPROVED** [review]
Approving with some comments we can discuss! I'm refactoring a bunch of things related to components on PR #518, so I'll merge this part around there as soon as merged to main

**Review by jurevans — APPROVED** [review]
Nice!

File: packages/components/src/Input/amountInput.tsx
Comment (emccorson) [reply packages/components/src/Input/amountInput.tsx:54]: I like the sound of that!

File: packages/components/src/Input/amountInput.tsx
Comment (emccorson) [reply packages/components/src/Input/amountInput.tsx:45]: Could you explain a bit more about this one please?

File: packages/components/src/Input/amountInput.tsx
Comment (mateuszjasiuk) [inline packages/components/src/Input/amountInput.tsx:49]: This is just a random thought, but I'm curious if mapping `ifs` predicates to Options like in this: https://gcanti.github.io/fp-ts/modules/Option.ts.html and then doing some kind of matching would reduce number of returns here :)

**Review by mateuszjasiuk — APPROVED** [review]
LGTM! Left two comments but they are more like a general thoughts

**Pull request merged**
